pipeline/load.py

```
@@ -0,0 +1,291 @@
+        return connection
+    except psycopg2.OperationalError:
+        logging.warning("The database %s doesn't exist", environ["DB_NAME"])
+        return None
+
+
+def get_cursor(connection: extensions.connection) -> Optional[extensions.cursor]:
+    """
+    Returns a the cursor for querying
+    the database from a connection.
+    """
+    try:
+        return connection.cursor()
+    except:
+        logging.warning("Unable to retrieve cursor for connection")
+        return None
+
+
+def get_id_from_country(country_name: str, cursor: extensions.cursor) -> int:
+    """
+    Retrieves the country_id of a 
+    given country from the database.
+    Returns -1 if it cannot be found.
+    """
+    search_query = f"SELECT country_id FROM country WHERE country_name = '{
+        country_name}';"
+    cursor.execute(search_query)
+    country_id = cursor.fetchone()
+    if not country_id:
+        logging.info("Unable to find id for %s", country_name)
+        return -1
+    return country_id[0]
+
+
+def get_id_from_artist(artist_name: str, cursor: extensions.cursor) -> int:
+    """
+    Retrieves the artist id of a 
+    given artist from the database.
+    Returns -1 if it cannot be found.
+    """
+    search_query = "SELECT artist_id FROM artist WHERE artist_name = %s;"
+    cursor.execute(search_query, (artist_name,))
+    artist_id = cursor.fetchone()
+    if not artist_id:
+        logging.info("Unable to find id for %s", artist_name)
+        return -1
+    return artist_id[0]
+
+
+def get_id_from_release_type(release_type: str, cursor: extensions.cursor) -> int:
+    """
+    Retrieves the release type id of a 
+    given release type.
+    Returns -1 if it cannot be found.
+    """
+    search_query = "SELECT type_id FROM type WHERE type_name = %s;"
+    cursor.execute(search_query, (release_type.lower(),))
+    type_id = cursor.fetchone()
+    if not type_id:
+        logging.info("Unable to find id for %s", release_type)
+        return -1
```

Maybe instead of returning -1 we should raise an exception or return None

---

After discussing with tech coaches, we have decided to change all return -1 to instead raise an error

---

I'm returning -1 to keep the variable output types the same. I'm not sure if raising an exception here is appropriate as there is no error to handle.